code.js

```diff
@@ -1,3 +1,35 @@
-function permutationSort(a) {
-    return 0;
+function permutationSort(a, lo) {
+    if (isSorted(a)) return 1;
+    let attempts = 0;
+    if (lo >= a.length-1) return 0;
+    for (let i = lo; i < a.length; i++) {
+	if (i != lo) {
+	    swap(a,lo,i);
```

Comment:
Does your function really try all permutations if you're swapping elements?

Reply:
Yep! I can add this line of code: 'console.log(a);' to two places to see each permutation. 1) After the swap function, 2) Right before the initial call to 'permSort(a,0)' so that we can see the original array. The pattern is easiest to see with a reverse sorted list.
permutationSort([4,3,2,1]):
[ 4, 3, 2, 1 ]
[ 4, 3, 1, 2 ]
[ 4, 2, 3, 1 ]
[ 4, 2, 1, 3 ]
[ 4, 1, 2, 3 ]
[ 4, 1, 3, 2 ]
[ 3, 4, 2, 1 ]
[ 3, 4, 1, 2 ]
[ 3, 2, 4, 1 ]
[ 3, 2, 1, 4 ]
[ 3, 1, 2, 4 ]
[ 3, 1, 4, 2 ]
[ 2, 3, 4, 1 ]
[ 2, 3, 1, 4 ]
[ 2, 4, 3, 1 ]
[ 2, 4, 1, 3 ]
[ 2, 1, 4, 3 ]
[ 2, 1, 3, 4 ]
[ 1, 3, 2, 4 ]
[ 1, 3, 4, 2 ]
[ 1, 2, 3, 4 ]
Total of 21 combinations, which falls short of 4! = 24, because this reverse-sorted list isn't actually the worst case input. We found the sorted list before we found [1,2,4,3] [1,4,3,2] and [1,4,2,3]. To see all combinations we try a list like this:
permuationSort([2,3,4,1]):
[ 2, 3, 4, 1 ]
[ 2, 3, 1, 4 ]
[ 2, 4, 3, 1 ]
[ 2, 4, 1, 3 ]
[ 2, 1, 4, 3 ]
[ 2, 1, 3, 4 ]
[ 3, 2, 4, 1 ]
[ 3, 2, 1, 4 ]
[ 3, 4, 2, 1 ]
[ 3, 4, 1, 2 ]
[ 3, 1, 4, 2 ]
[ 3, 1, 2, 4 ]
[ 4, 3, 2, 1 ]
[ 4, 3, 1, 2 ]
[ 4, 2, 3, 1 ]
[ 4, 2, 1, 3 ]
[ 4, 1, 2, 3 ]
[ 4, 1, 3, 2 ]
[ 1, 3, 4, 2 ]
[ 1, 3, 2, 4 ]
[ 1, 4, 3, 2 ]
[ 1, 4, 2, 3 ]
[ 1, 2, 4, 3 ]
[ 1, 2, 3, 4 ]
24 total permutations = 4!. 
permutationSort([2,3,4,5,6,7,8,1]) has a return value of 8! = 40320.